fn_ioc_parser/fn_ioc_parser/util/ioc_parser_helper.py

```diff
@@ -0,0 +1,152 @@
+                converter.close()
+                fake_file_handle.close()
+                os.unlink(temp_pdf_file.name)
+        return extracted_input
+
+    @classmethod
+    def extract_text_from_docx(cls, attachment_input):
+        """
+        Wrapper to convert bytes data in into .docx file and extracting the text data from .doc/.docx file
+        :param attachment_input:
+        :return: Text Data
+        """
+        extracted_input = u""
+        with tempfile.NamedTemporaryFile(mode="w+b", delete=False) as temp_doc_file:
+            try:
+                # Write and close temp file
+                temp_doc_file.write(attachment_input)
+                read_doc = Document(temp_doc_file)
+                for paragraph in read_doc.paragraphs:
+                    extracted_input += paragraph.text
+            finally:
+                os.unlink(temp_doc_file.name)
+        return extracted_input
+
+    @classmethod
+    def extract_text_from_xls_xlsx(cls, attachment_input):
+        """
+        Wrapper to convert bytes data in into .xls/.xlsx file and extracting the text data from .xls/.xlsx file
+        :param attachment_input:
+        :return: Text Data
+        """
+        extracted_input = ""
+        with tempfile.NamedTemporaryFile(mode="w+b", delete=False) as temp_doc_file:
+            try:
+                # Write and close temp file
+                temp_doc_file.write(attachment_input)
+                workbook_object = xlrd.open_workbook(temp_doc_file.name)
+                total_sheets = workbook_object.nsheets
+                for sheet_index in range(0, total_sheets):
+                    each_sheet_obj = workbook_object.sheet_by_index(sheet_index)
+                    total_rows_each_sheet = each_sheet_obj.nrows
+                    for row_no in range(0, total_rows_each_sheet):
+                        row_data_list = each_sheet_obj.row_values(row_no)
+                        for data in row_data_list:
+                            extracted_input += str(data) + " "
+            finally:
+                os.unlink(temp_doc_file.name)
+        return extracted_input
+
+    def _correct_ioc_value(self, kind, value):
+        _formatted_value = ''
+        if kind == 'uri':
+            if value.find('https://') != -1 or value.find('http://') != -1:
+                _formatted_value = value
+            else:
+                _formatted_value = "{0}{1}".format('https://', value)
+        else:
+            _formatted_value = value
+        return _formatted_value
+
+    def correct_iocs_format(self, ioc_parser_obj_list):
+        """
+        A wrapper to remove/correct invalid ioc's generated from IOC parser compatible to resilient artifacts
+        :param ioc_parser_obj_list:  IOC Parser Library result object List
+        :return:a python dict of result data
+        """
+        kind_map = {'uri': 'URL', 'IP': 'IP Address', 'md5': 'Malware MD5 Hash', 'sha1': 'Malware SHA-1 Hash',
```

Comment:
is this mapping still correct as you've added 'email subject'?

Reply:
yes its correct. 
IOC Parser can parse only email  id's in the given data, type of returned IOC will be always an email.
Data from resilient from might be anything related to email. 
now we needs to create an artifact for email, so when i checked available artifacts types  in resilient found email sender, email recipient. 
i can't able to make distinguish weather parsed email ioc is sender or recipient so chosen email body as artfifact type to create artifact from ioc data.